packages/blade/src/components/DatePicker/DatePicker.web.tsx

```diff
@@ -0,0 +1,297 @@
+            onDateChange(date);
+          }}
+          getMonthControlProps={(date) => {
+            return getControlProps(date);
+          }}
+          getYearControlProps={(date) => {
+            return getControlProps(date);
+          }}
+          getDayProps={(date) => {
+            return getControlProps(date);
+          }}
+          onMonthSelect={(date) => {
+            props?.onMonthSelect?.(date);
+            onDateChange(date);
+          }}
+          onYearSelect={(date) => {
+            props?.onYearSelect?.(date);
+            onDateChange(date);
+          }}
+          onNext={(date) => {
+            props?.onNext?.(date);
+            forceRerenderBottomSheet();
+          }}
+          onPrevious={(date) => {
+            props?.onPrevious?.(date);
+            forceRerenderBottomSheet();
+          }}
+          picker={_picker}
+          onPickerChange={(picker) => {
+            setPicker(() => picker);
+            forceRerenderBottomSheet();
+          }}
+        />
+        {isMobile ? null : <CalendarFooter onApply={handleApply} onCancel={handleCancel} />}
+      </BaseBox>
+    </>
+  );
+
+  return (
+    <BaseBox width="100%">
+      <DatePickerInput
+        selectionType={_selectionType}
+        date={controlledValue}
+        ref={referenceRef}
+        inputRef={refs.reference}
+        referenceProps={getReferenceProps()}
+        label={label as never}
+        labelPosition={labelPosition}
+        accessibilityLabel={accessibilityLabel}
+        size={size}
+        errorText={errorText}
+        helpText={helpText}
+        isDisabled={isDisabled}
+        isRequired={isRequired}
+        successText={successText}
+        validationState={validationState}
+      />
+      {isMobile ? (
+        <BottomSheet
+          snapPoints={[0.9, 0.9, 1]}
+          isOpen={controllableIsOpen}
+          onDismiss={() => {
+            handleCancel();
+          }}
+        >
+          <BottomSheetHeader title={isSingle ? 'Select Date' : 'Select Date Range'} />
+          <BottomSheetBody>
+            {content}
+            {!isSingle && (
+              <PresetSideBar
+                isMobile
+                presets={presets}
+                date={currentDate}
+                selectedPreset={selectedPreset}
+                onSelection={(preset) => {
+                  const presetValue = preset?.(currentDate);
+                  setControlledValue(presetValue);
+                  setSelectedPreset(presetValue);
+                }}
+              />
+            )}
+          </BottomSheetBody>
+          <BottomSheetFooter>
+            <CalendarFooter onCancel={handleCancel} onApply={handleApply} />
+          </BottomSheetFooter>
+        </BottomSheet>
+      ) : (
+        isMounted && (
+          <FloatingPortal>
+            <FloatingFocusManager
+              initialFocus={defaultInitialFocusRef}
+              context={context}
+              guards={true}
+            >
```

Comment:
I remember reading that FloatingPortal & FloatingFocusManager should always be mounted otherwise it may cause some issues

Reply:
🤔 Never heard about it. We do this in Popover,Tooltip etc components too. 

And is also documented/present on their official examples: https://codesandbox.io/p/sandbox/withered-shadow-ot38mm